lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -88,28 +94,65 @@ public ReportUserIdentityHandler() {
-                                                    usrId,
-                                                    Vot.P0.name(),
-                                                    Collections.emptyList(),
-                                                    false))
-                            .toList();
-            reportUserIdentityDataStore.createOrUpdate(reportUserIdentities);
+            ReportProcessingRequest reportProcessingRequest =
+                    objectMapper.readValue(inputStream, ReportProcessingRequest.class);
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            "Request received-" + reportProcessingRequest.toString()));
+            List<String> userIds = new ArrayList<>();
+            if ((!reportProcessingRequest.continueUserScan()
+                            && (reportProcessingRequest.tacticalStoreLastEvaluatedKey() == null))
+                    || reportProcessingRequest.continueUserScan()) {
+                LOGGER.info(
+                        LogHelper.buildLogMessage(
+                                "Retrieving userIds from tactical store db table."));
+
+                Map<String, AttributeValue> exclusiveStartKey =
+                        reportProcessingRequest.continueUserScan()
+                                ? reportProcessingRequest.tacticalStoreLastEvaluatedKey()
+                                : null;
+
+                do {
+                    TableScanResult<VcStoreItem> tableScanResult =
+                            vcStoreItemDataStore.getItems(exclusiveStartKey, ATTR_NAME_USER_ID);
+                    userIds.addAll(
+                            tableScanResult.items().stream()
+                                    .map(VcStoreItem::getUserId)
+                                    .distinct()
+                                    .toList());
 
+                    List<ReportUserIdentityItem> reportUserIdentities =
+                            userIds.stream()
+                                    .map(
+                                            usrId ->
+                                                    new ReportUserIdentityItem(
+                                                            usrId,
+                                                            null,
+                                                            0,
+                                                            Collections.emptyList(),
+                                                            false))
+                                    .toList();
+                    reportUserIdentityDataStore.createOrUpdate(reportUserIdentities);
+                    if (tableScanResult.lastEvaluatedKey() != null) {
+                        exclusiveStartKey = tableScanResult.lastEvaluatedKey();
+                        reportProcessingResult.tacticalStoreLastEvaluatedKey(exclusiveStartKey);
+                    }
+                } while (exclusiveStartKey != null);
```

Comment:
How fast does this scan execute? We might need a mechanism that allows us to stop processing if the lambda is about to time out, rather than powering through everything.

Reply:
Added a check for lambda timeout before starting a next pagination scan and gracefully exit if time left is less then 1sec.